src/lib/header-bidding/prebid/prebid.ts

```diff
@@ -172,13 +181,38 @@ class PrebidAdUnit {
 		this.code = advert.id;
 		this.bids = bids(advert.id, slot.sizes, pageTargeting);
 		this.mediaTypes = { banner: { sizes: slot.sizes } };
+		this.gpid = this.generateGpid(advert, slot, pageTargeting);
+		this.ortb2Imp = {
+			ext: {
+				gpid: this.gpid,
+				data: {
+					pbadslot: this.gpid,
+				},
+			},
+		};
 		advert.headerBiddingSizes = slot.sizes;
 		log('commercial', `PrebidAdUnit ${this.code}`, this.bids);
 	}
 
 	isEmpty() {
 		return this.code == null;
 	}
+	private generateGpid(
+		advert: Advert,
+		slot: HeaderBiddingSlot,
+		pageTargeting: PageTargeting,
+	): string {
+		const sectionName = Array.isArray(pageTargeting.sectionName)
```

Comment:
Are these ever arrays?

Reply:
Thanks have refactored to use window config